src/test/unit/gov/nist/javax/sip/parser/ims/PAssertedIdentityParserTest.java

```diff
@@ -41,5 +42,29 @@ public void testParser() {
 
         super.testParser(PAssertedIdentityParser.class,preferredID);
     }
+    
+    public void testInvalidHeaderParam() {
+    	
+        String[] preferredIDs =  {
+         "P-Asserted-Identity:sip:33296112233@provider.com;user=phone\n",
+        		
+         "P-Asserted-Identity: <sip:alice@atlanta.com>, <tel:+1-201-555-0123>;user=phone\n"
+
+        };
+
+        for (String preferredID :preferredIDs) {
+	        try {        
+	        	  
+	          
+	            System.out.println("preferredID:"+preferredID);
+	            HeaderParser hp = createParser(PAssertedIdentityParser.class, preferredID);
+	            hp.parse();
+	            fail("Header:"+ preferredID + " cannot be parse, wrong in pattern");
+	       
+		    } catch (java.text.ParseException ex) {
+		        
```

Comment:
try to add a positive assert here to enforce the expected behavior. Possibly include some check on the  Exception object. In this case something like assertTrue(e.getMessage().contains("doesn't allow parameters");